src/internal/GenericCodegen.scala

```diff
@@ -160,8 +161,10 @@ trait GenericCodegen extends BlockTraversal {
 
   def quote(x: Exp[Any]) : String = x match {
     case Const(s: String) => "\""+s.replace("\\", "\\\\").replace("\"", "\\\"").replace("\n", "\\n")+"\"" // TODO: more escapes?
-    case Const(f: Float) => f+"f"
     case Const(c: Char) => "'"+(""+c).replace("'", "\\'").replace("\n", "\\n")+"'"
+    case Const(f: Float) => "%1.10f".format(f) + "f"
```

Comment:
There's also some stuff in apache commons for doing general character escaping which covers most, if not all, of the special characters, so I added that in too.